scripts/src/osd/modules.lua

```diff
@@ -35,6 +39,30 @@ function addoptionsfromstring(str)
 	end
 end
 
+function addincluderootfromstring(str, EndPart)
+	if (str==nil) then return  end
+	for w in str:gmatch("%S+") do
+		if string.starts(w,"-I")==true then
+			if string.ends(w, EndPart) then
+				includedirs {
+					string.sub(w,3,string.len(w)-string.len(EndPart))
```

Comment:
You can use the `#` count operator on strings, and `:` method call syntax.  It’s easier to read like `w:sub(3, #w - #EndPart)`.